rpc/src/v1/impls/light/eth.rs

```diff
@@ -271,11 +270,8 @@ where
 		Ok(Default::default())
 	}
 
-	fn gas_price(&self) -> Result<U256> {
-		Ok(self.cache.lock().gas_price_corpus()
-			.and_then(|c| c.percentile(self.gas_price_percentile).cloned())
-			.map(U256::from)
-			.unwrap_or_else(Default::default))
+	fn gas_price(&self) -> BoxFuture<U256> {
+		Box::new(self.fetcher().gas_price())
```

Comment:
Note, `dispatch::light::fetch_gas_price_corpus` will read cache first before trying to perform a N/W request.

However, it will cost a few extra `Arc::clone` to construct `LightFetch` instead of directly call the cache